test/unit/Sql/Predicate/PredicateTest.php

```diff
@@ -276,4 +281,59 @@ public function testLiteral()
             $predicate->getExpressionData()
         );
     }
+
+    public function testCanCreateExpressionsWithoutAnyBoundSqlParameters(): void
+    {
+        $where1 = new Predicate();
+
+        $where1->expression('some_expression()');
+
+        self::assertSame(
+            'SELECT "a_table".* FROM "a_table" WHERE (some_expression())',
+            $this->makeSqlString($where1)
+        );
+    }
+
+    public function testWillBindSqlParametersToExpressionsWithGivenParameter(): void
+    {
+        $where = new Predicate();
+
+        $where->expression('some_expression(?)', null);
+
+        self::assertSame(
+            'SELECT "a_table".* FROM "a_table" WHERE (some_expression(\'\'))',
```

Comment:
Note: not interpolating `NULL` here, because that's what the Sql92 platform dictates.

Reply:
That was my question.

I guess the bigger question I have is that the value is called `$parameters`, and type-hinted as an array. Passing `null` feels like you are saying "omit the parameters", versus "cast to a string".

The difficulty here, of course, is that the value was being passed verbatim to the `Expression` constructor, which allows `null`, and this would normally be cast as a string. I'm honestly not sure what the right answer is here; the previous behavior feels more like a bug than anything.

Reply:
Agreed: the problem here is that `null` being passed in is certainly an existing use-case, which we can't change at this stage.

I also had to scratch my eyes when I saw `null` being converted to `''`, but that's already the current use-case.